h2o-algos/src/main/java/hex/gam/GamSplines/ThinPlateRegressionUtils.java

```diff
@@ -0,0 +1,346 @@
+        knotsDemean[predInd][index] = standardizeTPSmoothers 
+                ? (knots[predInd][index]-gamColMeanRaw[predInd])*oneOColStd[predInd] 
+                : (knots[predInd][index]-gamColMeanRaw[predInd]);
+      }
+    
+    double[][] starT = new double[numKnots][M];
+    for (int rowInd = 0; rowInd < numKnots; rowInd++) {
+      for (int polyBasisInd = 0; polyBasisInd < M; polyBasisInd++) {
+        Integer[] oneBasis = polyBasisDegree.get(polyBasisInd);
+        double polyBasisVal = 1.0;
+        for (int predInd = 0; predInd < d; predInd++) {
+          polyBasisVal *= Math.pow(knotsDemean[predInd][rowInd], oneBasis[predInd]);
+        }
+        starT[rowInd][polyBasisInd] = polyBasisVal;
+      }
+    }
+    return starT;
+  }
+  
+  public static void fillRowOneValue(NewChunk[] newChk, int colWidth, double fillValue) {
+    for (int colInd = 0; colInd < colWidth; colInd++)
+      newChk[colInd].addNum(fillValue);
+  }
+
+  public static void fillRowArray(NewChunk[] newChk, int colWidth, double[] fillValue) {
+    for (int colInd = 0; colInd < colWidth; colInd++)
+      newChk[colInd].addNum(fillValue[colInd]);
+  }
+
+  public static boolean checkRowNA(Chunk[] chk, int rowIndex) {
+    int numCol = chk.length;
+    for (int colIndex = 0; colIndex < numCol; colIndex++) {
+      if (Double.isNaN(chk[colIndex].atd(rowIndex)))
+        return true;
+    }
+    return false;
+  }
+
+  public static boolean checkFrameRowNA(Frame chk, long rowIndex) {
+    int numCol = chk.numCols();
+    for (int colIndex = 0; colIndex < numCol; colIndex++) {
+      if (Double.isNaN(chk.vec(colIndex).at(rowIndex)))
+        return true;
+    }
+    return false;
+  }
+  
+  public static String genThinPlateNameStart(GAMParameters parms, int gamColIndex) {
+    StringBuffer colNameStub = new StringBuffer();
+    for (int gColInd = 0; gColInd < parms._gam_columns_sorted[gamColIndex].length; gColInd++) {
+      colNameStub.append(parms._gam_columns_sorted[gamColIndex][gColInd]);
+      colNameStub.append("_");
+    }
+    colNameStub.append(parms._bs_sorted[gamColIndex]);
+    colNameStub.append("_");
+    return colNameStub.toString();
+  }
+  
+  public static String[] extractColNames(String[] src, int srcStart, int destStart, int length) {
+    String[] distanceColNames = new String[length];  // exclude the polynomial basis names
+    System.arraycopy(src, srcStart, distanceColNames, destStart, length);
+    return distanceColNames;
+  }
+  
+  public static int[][] convertList2Array(List<Integer[]> list2Convert, int M, int d) {
+    int[][] polyBasisArr = new int[M][d];
+    for (int index = 0; index < M; index++) {
+      List<Integer> oneList = Arrays.asList(list2Convert.get(index));
+      polyBasisArr[index] = oneList.stream().mapToInt(Integer::intValue).toArray();
+    }
+    return polyBasisArr;
+  }
+  
+  // generate knots for TP smoothers.  Sort the first predictor column, take quantiles out of the sorted
```

Comment:
Same here, it is just cosmetics, however, Javadoc can really help read the documentation.

Reply:
@maurever  I changed both of them to java docs.  Thanks.